bitcoin_utils.py

```diff
@@ -30,7 +31,10 @@ def deser_txid(txid: str):
     return bytes.fromhex(dixt)
 
 
-def deser_compact_size(f):
+def deser_compact_size(f: BytesIO):
+    if (len(f.read(1)) == 0):
+        return 0 # end of stream
+    f.seek(-1, os.SEEK_CUR) # rewind
```

Comment:
If I'm understanding this, it's to catch when the stream passed is empty? Using `os` seems a bit overkill, is there a method on `BytesIO` we could use instead?

Reply:
I think you are referring to a method like `peek` which `io.BufferedReader` has but [BytesIO](https://docs.python.org/3/library/io.html?highlight=bytesio#io.BytesIO) doesn't support peeking. I can implement this in a different way using a temporary buffer which would remove the need for me to "rewind" the stream

Reply:
Looks like `BytesIO` has a `.getbuffer()` function , which has an `nbytes` property. so we could do something like `if f.getbuffer().nbytes == 0`? feels a bit more elegant than reading and then rewinding

Reply:
Done!